src/garage/replay_buffer/meta_replay_buffer.py

```diff
@@ -0,0 +1,145 @@
+"""A replay buffer memory for meta RL."""
+
+import numpy as np
+
+
+class MetaReplayBuffer:
+    """This class implements MetaReplayBuffer.
+
+    It stores information of each sample in separate numpy arrays.
+
+    Args:
+        max_replay_buffer_size (int): Maximum buffer size.
+        observation_dim (numpy.ndarray): Dimension of observations.
+        action_dim (numpy.ndarray): Dimension of actions.
+
+    """
+
+    def __init__(self, max_replay_buffer_size, observation_dim, action_dim):
+
+        self._observation_dim = observation_dim
+        self._action_dim = action_dim
+        self._max_replay_buffer_size = max_replay_buffer_size
+        self._observations = np.zeros(
+            (max_replay_buffer_size, observation_dim))
+        self._next_obs = np.zeros((max_replay_buffer_size, observation_dim))
+        self._actions = np.zeros((max_replay_buffer_size, action_dim))
+        self._rewards = np.zeros((max_replay_buffer_size, 1))
+        self._terminals = np.zeros((max_replay_buffer_size, 1), dtype='uint8')
+        self._top = 0
+        self._size = 0
+        self._episode_starts = []
+        self._cur_episode_start = 0
+
+    def add_sample(self, observation, action, reward, terminal,
+                   next_observation):
+        """Add a sample to the buffer.
+
+        Args:
+            observation (numpy.ndarray): Observation.
+            action (numpy.ndarray): Action.
+            reward (float): Reward.
+            terminal (bool): Terminal state.
+            next_observation (numpy.ndarray): Next obseravation.
+
+        """
+        self._observations[self._top] = observation
+        self._actions[self._top] = action
+        self._rewards[self._top] = reward
+        self._terminals[self._top] = terminal
+        self._next_obs[self._top] = next_observation
+        self._advance()
+
+    def add_path(self, path):
+        """Add a path to buffer.
+
+        Args:
+            path (dict): Dictionary containing path information.
+
+        """
+        for _, (obs, action, reward, next_obs, terminal) in enumerate(
+                zip(path['observations'], path['actions'], path['rewards'],
+                    path['next_observations'], path['terminals'])):
+            self.add_sample(obs, action, reward, terminal, next_obs)
+        self.terminate_episode()
+
+    def size(self):
+        """Get size of buffer.
+
+        Returns:
+            int: Current size of buffer.
+
+        """
+        return self._size
+
+    def _advance(self):
+        """Increment size of buffer."""
+        self._top = (self._top + 1) % self._max_replay_buffer_size
```

Comment:
i think you should document in the class docstring that this is a FIFO replay buffer